lib/ansible/modules/windows/win_service.py

```diff
@@ -113,6 +113,47 @@
     - A newly created service will default to C(LocalSystem).
     type: str
     version_added: '2.3'
+  recovery:
+    description:
+      - A dictionary representing the recovery actions you want to set.
+    suboptions:
+      reset_fail_count_after:
+        description:
+          - Reset failure counter after this number of seconds.
+        type: int
+      on_first_failure:
```

Comment:
Was talking this over with @jborean93 after I went looking at the underlying Win32 API for this, and the way `sc.exe` sets it as well, and we were thinking it might be better if, rather than mimic'ing the SCM UI for this, if it took a list of actions. That allows for shorter descriptions, like:
```
recovery_reset_interval: 300
recovery_actions:
- action: restart
  delay: 60000
```

but it means the registry side of things needs to be a little more dynamic, since cActions would vary with the size of the list. It also makes the return value a little simpler, since it's just a list. Thoughts?

Reply:
The recovery_actions been passed as a list seems much nicer indeed, and not forget shorter. Although when you think about the fact that a list is in theory not ordered (even if I've seen that on many occasions the order doesn't change), would it be a good idea to do it? Are there assurances that can be given in order to be sure a list stays in order?

Reply:
A list should be ordered, it's a dict that doesn't have a guaranteed order to the keys.

Reply:
I do agree that a dynamic list would be easier to document and use.
I'll check what I can do to make it more dynamic. It also makes sense.

It would also make it less "bomby" in case of parameter not passed.

But changing this, would require me to redo a bunch of tests for idempotency, so if it happens, it'll take me some time.

Reply:
@nitzmahone @jborean93 so changing it to a list is gonna take more time than I thought, not sure if I'll be able to make for the code freeze of Ansible 2.9.
Do you think this can already be a working solution for 2.9 and we can iterate on it for the next releases?

Reply:
@rnsc sorry for the delay, I would prefer not to implement a change then change the UI for those options in the future. The freeze isn't until the end of August, if you think you won't be able to make it let me know and I'll try to pick up where you've left off and see if we can get it in before that timeframe.

Reply:
Hello @nitzmahone and @jborean93 
I implemented the changes you wanted to see, let me know if it's better now :)